src/Datadog.NativeLoader/dllmain.cpp

```diff
@@ -0,0 +1,176 @@
+    {
+        profiler_path = GetEnvironmentValue(WStr("CORECLR_PROFILER_PATH_64"));
+    }
+    if (profiler_path.length() == 0)
+    {
+        profiler_path = GetEnvironmentValue(WStr("COR_PROFILER_PATH_64"));
+    }
+#else
+    if (profiler_path.length() == 0)
+    {
+        profiler_path = GetEnvironmentValue(WStr("CORECLR_PROFILER_PATH_32"));
+    }
+    if (profiler_path.length() == 0)
+    {
+        profiler_path = GetEnvironmentValue(WStr("COR_PROFILER_PATH_32"));
+    }
+#endif
+
+    return profiler_path;
+}
+
+extern "C"
+{
+    BOOL STDMETHODCALLTYPE DllMain(HMODULE hModule, DWORD ul_reason_for_call, LPVOID lpReserved)
+    {
+        // Perform actions based on the reason for calling.
+        switch (ul_reason_for_call)
+        {
+            case DLL_PROCESS_ATTACH:
+            {
+                // Initialize once for each new process.
+                // Return FALSE to fail DLL load.
+
+                Debug("DllMain - DLL_PROCESS_ATTACH");
+
+                dispatcher = new DynamicDispatcher();
+
+                WSTRING profiler_path = GetProfilerPath();
+                std::string path =
+                    std::filesystem::path(profiler_path).remove_filename().append(conf_filename).string();
+
+                std::unordered_map<std::string, bool> guidBoolMap;
+
+                std::ifstream t;
+                t.open(path);
+                while (t)
+                {
+                    std::string line;
+                    std::getline(t, line);
+                    line = Trim(line);
+                    if (line.length() != 0)
+                    {
+                        Debug(line);
+
+                        if (line.substr(0, 1) == "#")
+                        {
+                            continue;
+                        }
+
+                        size_t delimiter = line.find("=");
+                        std::string filepath = line.substr(delimiter + 1);
+                        std::string clsid = line.substr(0, delimiter);
+
+                        filepath = std::filesystem::path(filepath).replace_extension(dynExtension).string();
+                        if (std::filesystem::exists(filepath))
+                        {
+                            guidBoolMap[clsid] = true;
+                            std::unique_ptr<DynamicInstance> instance = std::make_unique<DynamicInstance>(filepath, clsid);
+                            dispatcher->Add(instance);
+                            auto envVal = SetEnvironmentValue(WStr("PROFID_") + ToWSTRING(clsid), ToWSTRING(filepath));
+                            Debug("SetEnvVal: ", envVal, "; ", WStr("PROFID_") + ToWSTRING(clsid), "=", ToWSTRING(filepath));
+                        }
+                        else if (guidBoolMap.find(clsid) == guidBoolMap.end())
+                        {
+                            guidBoolMap[clsid] = false;
+                        }
+                    }
+                }
+                t.close();
+
```

Comment:
wouldn't it worth having the parsing of the conf file in util file and add unit tests on it? (will be able to test the Guid parser also)

Reply:
Sure, I'm just trying to make it work right now XD

Reply:
👍

Reply:
I moved this logic to the DynamicDispatcher class. I have to figure out the testing part later.